src/test/java/seedu/duke/command/DatabaseTest.java

```diff
@@ -0,0 +1,140 @@
+package seedu.duke.command;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.duke.exceptions.InvalidModuleException;
+import seedu.duke.exceptions.InvalidModuleMappingException;
+import seedu.duke.exceptions.InvalidUniversityException;
+import seedu.duke.exceptions.ModuleNotFoundException;
+import seedu.duke.exceptions.UniversityNotFoundException;
+import seedu.duke.module.Module;
+import seedu.duke.module.ModuleMapping;
+import seedu.duke.university.University;
+
+public class DatabaseTest {
+    @Test
+    void addModuleMapping_validModuleMapping_correctString()
+            throws InvalidUniversityException, InvalidModuleException, InvalidModuleMappingException {
+        University partnerUniversity = new University("Zhejiang University", "China");
+        Module partnerUniversityModule = new Module("21121340", "Computer Networks", "2.5", partnerUniversity);
+        University nus = new University("NUS", "Singapore");
+        Module nusModule = new Module("CS2105", "Intro to Computer Networks", "4", nus);
+        ModuleMapping moduleMapping = new ModuleMapping(partnerUniversityModule, nusModule);
+
+        Database.addModuleMapping(moduleMapping);
+
+        assertEquals(
+                "21121340 Computer Networks 2.5MCs in Zhejiang University | CS2105 Intro to Computer "
+                        + "Networks 4MCs in NUS",
+                Database.getModuleMappings().get(Database.getModuleMappings().size() - 1).toString());
+    }
+
+    @Test
+    void addUniversity_validUniversity_correctString() throws InvalidUniversityException {
+        University nus = new University("NUS", "Singapore");
+
+        Database.addUniversity(nus);
+
+        assertEquals("NUS", Database.getUniversities().get(0).toString());
+    }
+
+    @Test
+    void findPuMapping_validModule_correctList() throws ModuleNotFoundException {
+        DatabaseStorage.loadDatabase();
+
+        assertEquals(
+                "C3130 Information Security 5MCs in Aalto University | CS2107 Introduction to Information "
+                        + "Se 4MCs in NUS",
+                Database.findPuMapping("C3130", "Aalto University").toString());
+    }
+
+    @Test
+    void findPuMapping_invalidModule_throwsException() throws ModuleNotFoundException {
+        DatabaseStorage.loadDatabase();
+
+        assertThrows(ModuleNotFoundException.class, () -> Database.findPuMapping("CS2113", "ABCDEFG").toString());
+    }
+
+    @Test
+    void findNusMapping_validModule_correctList() throws ModuleNotFoundException {
+        DatabaseStorage.loadDatabase();
+
+        assertEquals(
+                "[COM162 Object Oriented Design and Programming with Java 15MCs in The University of "
```

Comment:
In general we avoid hard-coded values when testing. But if the expected value is not volatile (e.g. not mutating due to changes in database, API, or other various reasons), then it should be fine.